File: VB/4PosBackOffice.NET/AxMonthViewArray.vb
'UPGRADE_WARNING: The entire project must be compiled once before a form with an ActiveX Control Array can be displayed

Imports System.ComponentModel

<ProvideProperty("Index",GetType(System.Windows.Forms.MonthCalendar))> Public Class AxMonthViewArray
	Inherits Microsoft.VisualBasic.Compatibility.VB6.BaseOcxArray
	Implements IExtenderProvider

	Public Sub New()
		MyBase.New()
	End Sub

	Public Sub New(ByVal Container As IContainer)
		MyBase.New(Container)
	End Sub

	Public Shadows Event [DateClick] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_DateClickEvent)
	Public Shadows Event [DateDblClick] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_DateDblClickEvent)
	Public Shadows Event [GetDayBold] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_GetDayBoldEvent)
	Public Shadows Event [SelChange] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_SelChangeEvent)
	Public Shadows Event [ClickEvent] (ByVal sender As System.Object, ByVal e As System.EventArgs)
	Public Shadows Event [DblClick] (ByVal sender As System.Object, ByVal e As System.EventArgs)
	Public Shadows Event [KeyDown] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyDown)
	Public Shadows Event [KeyUpEvent] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyUpEvent)
	Public Shadows Event [KeyPress] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyPress)
	Public Shadows Event [MouseDown] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseDown)
	Public Shadows Event [MouseMoveEvent] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseMoveEvent)
	Public Shadows Event [MouseUpEvent] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseUpEvent)
	Public Shadows Event [OLEStartDrag] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEStartDragEvent)
	Public Shadows Event [OLEGiveFeedback] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEvent)
	Public Shadows Event [OLESetData] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLESetDataEvent)
	Public Shadows Event [OLECompleteDrag] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLECompleteDragEvent)
	Public Shadows Event [OLEDragOver] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEDragOverEvent)
	Public Shadows Event [OLEDragDrop] (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEDragDropEvent)

	<System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)> Public Function CanExtend(ByVal target As Object) As Boolean Implements IExtenderProvider.CanExtend
		If TypeOf target Is System.Windows.Forms.MonthCalendar Then
			Return BaseCanExtend(target)
		End If
	End Function

	Public Function GetIndex(ByVal o As System.Windows.Forms.MonthCalendar) As Short
		Return BaseGetIndex(o)
	End Function

	<System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)> Public Sub SetIndex(ByVal o As System.Windows.Forms.MonthCalendar, ByVal Index As Short)
		BaseSetIndex(o, Index)
	End Sub

	<System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)> Public Function ShouldSerializeIndex(ByVal o As System.Windows.Forms.MonthCalendar) As Boolean
		Return BaseShouldSerializeIndex(o)
	End Function

	<System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)> Public Sub ResetIndex(ByVal o As System.Windows.Forms.MonthCalendar)
		BaseResetIndex(o)
	End Sub

	Default Public ReadOnly Property Item(ByVal Index As Short) As System.Windows.Forms.MonthCalendar
		Get
			Item = CType(BaseGetItem(Index), System.Windows.Forms.MonthCalendar)
		End Get
	End Property

	Protected Overrides Function GetControlInstanceType() As System.Type
		Return GetType(System.Windows.Forms.MonthCalendar)
	End Function

	Protected Overrides Sub HookUpControlEvents(ByVal o As Object)
		Dim ctl As System.Windows.Forms.MonthCalendar = CType(o, System.Windows.Forms.MonthCalendar)
		MyBase.HookUpControlEvents(o)
		If Not DateClickEvent Is Nothing Then
			AddHandler ctl.DateClick, New AxMSComCtl2.DMonthViewEvents_DateClickEventHandler(AddressOf HandleDateClick)
		End If
		If Not DateDblClickEvent Is Nothing Then
			AddHandler ctl.DateDblClick, New AxMSComCtl2.DMonthViewEvents_DateDblClickEventHandler(AddressOf HandleDateDblClick)
		End If
		If Not GetDayBoldEvent Is Nothing Then
			AddHandler ctl.GetDayBold, New AxMSComCtl2.DMonthViewEvents_GetDayBoldEventHandler(AddressOf HandleGetDayBold)
		End If
		If Not SelChangeEvent Is Nothing Then
			AddHandler ctl.SelChange, New AxMSComCtl2.DMonthViewEvents_SelChangeEventHandler(AddressOf HandleSelChange)
		End If
		If Not ClickEventEvent Is Nothing Then
			AddHandler ctl.ClickEvent, New System.EventHandler(AddressOf HandleClickEvent)
		End If
		If Not DblClickEvent Is Nothing Then
			AddHandler ctl.DblClick, New System.EventHandler(AddressOf HandleDblClick)
		End If
		If Not KeyDownEvent Is Nothing Then
			AddHandler ctl.KeyDown, New AxMSComCtl2.DMonthViewEvents_KeyDownHandler(AddressOf HandleKeyDown)
		End If
		If Not KeyUpEventEvent Is Nothing Then
			AddHandler ctl.KeyUpEvent, New AxMSComCtl2.DMonthViewEvents_KeyUpEventHandler(AddressOf HandleKeyUpEvent)
		End If
		If Not KeyPressEvent Is Nothing Then
			AddHandler ctl.KeyPress, New AxMSComCtl2.DMonthViewEvents_KeyPressHandler(AddressOf HandleKeyPress)
		End If
		If Not MouseDownEvent Is Nothing Then
			AddHandler ctl.MouseDown, New AxMSComCtl2.DMonthViewEvents_MouseDownHandler(AddressOf HandleMouseDown)
		End If
		If Not MouseMoveEventEvent Is Nothing Then
			AddHandler ctl.MouseMoveEvent, New AxMSComCtl2.DMonthViewEvents_MouseMoveEventHandler(AddressOf HandleMouseMoveEvent)
		End If
		If Not MouseUpEventEvent Is Nothing Then
			AddHandler ctl.MouseUpEvent, New AxMSComCtl2.DMonthViewEvents_MouseUpEventHandler(AddressOf HandleMouseUpEvent)
		End If
		If Not OLEStartDragEvent Is Nothing Then
			AddHandler ctl.OLEStartDrag, New AxMSComCtl2.DMonthViewEvents_OLEStartDragEventHandler(AddressOf HandleOLEStartDrag)
		End If
		If Not OLEGiveFeedbackEvent Is Nothing Then
			AddHandler ctl.OLEGiveFeedback, New AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEventHandler(AddressOf HandleOLEGiveFeedback)
		End If
		If Not OLESetDataEvent Is Nothing Then
			AddHandler ctl.OLESetData, New AxMSComCtl2.DMonthViewEvents_OLESetDataEventHandler(AddressOf HandleOLESetData)
		End If
		If Not OLECompleteDragEvent Is Nothing Then
			AddHandler ctl.OLECompleteDrag, New AxMSComCtl2.DMonthViewEvents_OLECompleteDragEventHandler(AddressOf HandleOLECompleteDrag)
		End If
		If Not OLEDragOverEvent Is Nothing Then
			AddHandler ctl.OLEDragOver, New AxMSComCtl2.DMonthViewEvents_OLEDragOverEventHandler(AddressOf HandleOLEDragOver)
		End If
		If Not OLEDragDropEvent Is Nothing Then
			AddHandler ctl.OLEDragDrop, New AxMSComCtl2.DMonthViewEvents_OLEDragDropEventHandler(AddressOf HandleOLEDragDrop)
		End If
	End Sub

	Private Sub HandleDateClick (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_DateClickEvent) 
		RaiseEvent [DateClick] (sender, e)
	End Sub

	Private Sub HandleDateDblClick (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_DateDblClickEvent) 
		RaiseEvent [DateDblClick] (sender, e)
	End Sub

	Private Sub HandleGetDayBold (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_GetDayBoldEvent) 
		RaiseEvent [GetDayBold] (sender, e)
	End Sub

	Private Sub HandleSelChange (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_SelChangeEvent) 
		RaiseEvent [SelChange] (sender, e)
	End Sub

	Private Sub HandleClickEvent (ByVal sender As System.Object, ByVal e As System.EventArgs) 
		RaiseEvent [ClickEvent] (sender, e)
	End Sub

	Private Sub HandleDblClick (ByVal sender As System.Object, ByVal e As System.EventArgs) 
		RaiseEvent [DblClick] (sender, e)
	End Sub

	Private Sub HandleKeyDown (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyDown) 
		RaiseEvent [KeyDown] (sender, e)
	End Sub

	Private Sub HandleKeyUpEvent (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyUpEvent) 
		RaiseEvent [KeyUpEvent] (sender, e)
	End Sub

	Private Sub HandleKeyPress (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_KeyPress) 
		RaiseEvent [KeyPress] (sender, e)
	End Sub

	Private Sub HandleMouseDown (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseDown) 
		RaiseEvent [MouseDown] (sender, e)
	End Sub

	Private Sub HandleMouseMoveEvent (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseMoveEvent) 
		RaiseEvent [MouseMoveEvent] (sender, e)
	End Sub

	Private Sub HandleMouseUpEvent (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_MouseUpEvent) 
		RaiseEvent [MouseUpEvent] (sender, e)
	End Sub

	Private Sub HandleOLEStartDrag (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEStartDragEvent) 
		RaiseEvent [OLEStartDrag] (sender, e)
	End Sub

	Private Sub HandleOLEGiveFeedback (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEvent) 
		RaiseEvent [OLEGiveFeedback] (sender, e)
	End Sub

	Private Sub HandleOLESetData (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLESetDataEvent) 
		RaiseEvent [OLESetData] (sender, e)
	End Sub

	Private Sub HandleOLECompleteDrag (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLECompleteDragEvent) 
		RaiseEvent [OLECompleteDrag] (sender, e)
	End Sub

	Private Sub HandleOLEDragOver (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEDragOverEvent) 
		RaiseEvent [OLEDragOver] (sender, e)
	End Sub

	Private Sub HandleOLEDragDrop (ByVal sender As System.Object, ByVal e As AxMSComCtl2.DMonthViewEvents_OLEDragDropEvent) 
		RaiseEvent [OLEDragDrop] (sender, e)
	End Sub

End Class

File: C-Sharp/AxMonthViewArray.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
//UPGRADE_WARNING: The entire project must be compiled once before a form with an ActiveX Control Array can be displayed

using System.ComponentModel;
namespace _4PosBackOffice.NET
{

	[ProvideProperty("Index", typeof(System.Windows.Forms.MonthCalendar))]
	public class AxMonthViewArray : Microsoft.VisualBasic.Compatibility.VB6.BaseOcxArray, IExtenderProvider
	{

		public AxMonthViewArray() : base()
		{
		}

		public AxMonthViewArray(IContainer Container) : base(Container)
		{
		}

		public new event DateClickEventHandler DateClick;
		public new delegate void DateClickEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_DateClickEvent e);
		public new event DateDblClickEventHandler DateDblClick;
		public new delegate void DateDblClickEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_DateDblClickEvent e);
		public new event GetDayBoldEventHandler GetDayBold;
		public new delegate void GetDayBoldEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_GetDayBoldEvent e);
		public new event SelChangeEventHandler SelChange;
		public new delegate void SelChangeEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_SelChangeEvent e);
		public new event ClickEventEventHandler ClickEvent;
		public new delegate void ClickEventEventHandler(System.Object sender, System.EventArgs e);
		public new event DblClickEventHandler DblClick;
		public new delegate void DblClickEventHandler(System.Object sender, System.EventArgs e);
		public new event KeyDownEventHandler KeyDown;
		public new delegate void KeyDownEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyDown e);
		public new event KeyUpEventEventHandler KeyUpEvent;
		public new delegate void KeyUpEventEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyUpEvent e);
		public new event KeyPressEventHandler KeyPress;
		public new delegate void KeyPressEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyPress e);
		public new event MouseDownEventHandler MouseDown;
		public new delegate void MouseDownEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseDown e);
		public new event MouseMoveEventEventHandler MouseMoveEvent;
		public new delegate void MouseMoveEventEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseMoveEvent e);
		public new event MouseUpEventEventHandler MouseUpEvent;
		public new delegate void MouseUpEventEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseUpEvent e);
		public new event OLEStartDragEventHandler OLEStartDrag;
		public new delegate void OLEStartDragEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEStartDragEvent e);
		public new event OLEGiveFeedbackEventHandler OLEGiveFeedback;
		public new delegate void OLEGiveFeedbackEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEvent e);
		public new event OLESetDataEventHandler OLESetData;
		public new delegate void OLESetDataEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLESetDataEvent e);
		public new event OLECompleteDragEventHandler OLECompleteDrag;
		public new delegate void OLECompleteDragEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLECompleteDragEvent e);
		public new event OLEDragOverEventHandler OLEDragOver;
		public new delegate void OLEDragOverEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEDragOverEvent e);
		public new event OLEDragDropEventHandler OLEDragDrop;
		public new delegate void OLEDragDropEventHandler(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEDragDropEvent e);

		[System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
		public bool CanExtend(object target)
		{
			if (target is System.Windows.Forms.MonthCalendar) {
				return BaseCanExtend(target);
			}
		}

		public short GetIndex(System.Windows.Forms.MonthCalendar o)
		{
			return BaseGetIndex(o);
		}

		[System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
		public void SetIndex(System.Windows.Forms.MonthCalendar o, short Index)
		{
			BaseSetIndex(o, Index);
		}

		[System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
		public bool ShouldSerializeIndex(System.Windows.Forms.MonthCalendar o)
		{
			return BaseShouldSerializeIndex(o);
		}

		[System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
		public void ResetIndex(System.Windows.Forms.MonthCalendar o)
		{
			BaseResetIndex(o);
		}

		public System.Windows.Forms.MonthCalendar this[short Index] {
			get { return (System.Windows.Forms.MonthCalendar)BaseGetItem(Index); }
		}

		protected override System.Type GetControlInstanceType()
		{
			return typeof(System.Windows.Forms.MonthCalendar);
		}

		protected override void HookUpControlEvents(object o)
		{
			System.Windows.Forms.MonthCalendar ctl = (System.Windows.Forms.MonthCalendar)o;
			base.HookUpControlEvents(o);
			if ((DateClickEvent != null)) {
				ctl.DateClick += new AxMSComCtl2.DMonthViewEvents_DateClickEventHandler(HandleDateClick);
			}
			if ((DateDblClickEvent != null)) {
				ctl.DateDblClick += new AxMSComCtl2.DMonthViewEvents_DateDblClickEventHandler(HandleDateDblClick);
			}
			if ((GetDayBoldEvent != null)) {
				ctl.GetDayBold += new AxMSComCtl2.DMonthViewEvents_GetDayBoldEventHandler(HandleGetDayBold);
			}
			if ((SelChangeEvent != null)) {
				ctl.SelChange += new AxMSComCtl2.DMonthViewEvents_SelChangeEventHandler(HandleSelChange);
			}
			if ((ClickEventEvent != null)) {
				ctl.ClickEvent += new System.EventHandler(HandleClickEvent);
			}
			if ((DblClickEvent != null)) {
				ctl.DblClick += new System.EventHandler(HandleDblClick);
			}
			if ((KeyDownEvent != null)) {
				ctl.KeyDown += new AxMSComCtl2.DMonthViewEvents_KeyDownHandler(HandleKeyDown);
			}
			if ((KeyUpEventEvent != null)) {
				ctl.KeyUpEvent += new AxMSComCtl2.DMonthViewEvents_KeyUpEventHandler(HandleKeyUpEvent);
			}
			if ((KeyPressEvent != null)) {
				ctl.KeyPress += new AxMSComCtl2.DMonthViewEvents_KeyPressHandler(HandleKeyPress);
			}
			if ((MouseDownEvent != null)) {
				ctl.MouseDown += new AxMSComCtl2.DMonthViewEvents_MouseDownHandler(HandleMouseDown);
			}
			if ((MouseMoveEventEvent != null)) {
				ctl.MouseMoveEvent += new AxMSComCtl2.DMonthViewEvents_MouseMoveEventHandler(HandleMouseMoveEvent);
			}
			if ((MouseUpEventEvent != null)) {
				ctl.MouseUpEvent += new AxMSComCtl2.DMonthViewEvents_MouseUpEventHandler(HandleMouseUpEvent);
			}
			if ((OLEStartDragEvent != null)) {
				ctl.OLEStartDrag += new AxMSComCtl2.DMonthViewEvents_OLEStartDragEventHandler(HandleOLEStartDrag);
			}
			if ((OLEGiveFeedbackEvent != null)) {
				ctl.OLEGiveFeedback += new AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEventHandler(HandleOLEGiveFeedback);
			}
			if ((OLESetDataEvent != null)) {
				ctl.OLESetData += new AxMSComCtl2.DMonthViewEvents_OLESetDataEventHandler(HandleOLESetData);
			}
			if ((OLECompleteDragEvent != null)) {
				ctl.OLECompleteDrag += new AxMSComCtl2.DMonthViewEvents_OLECompleteDragEventHandler(HandleOLECompleteDrag);
			}
			if ((OLEDragOverEvent != null)) {
				ctl.OLEDragOver += new AxMSComCtl2.DMonthViewEvents_OLEDragOverEventHandler(HandleOLEDragOver);
			}
			if ((OLEDragDropEvent != null)) {
				ctl.OLEDragDrop += new AxMSComCtl2.DMonthViewEvents_OLEDragDropEventHandler(HandleOLEDragDrop);
			}
		}

		private void HandleDateClick(System.Object sender, AxMSComCtl2.DMonthViewEvents_DateClickEvent e)
		{
			if (DateClick != null) {
				DateClick(sender, e);
			}
		}

		private void HandleDateDblClick(System.Object sender, AxMSComCtl2.DMonthViewEvents_DateDblClickEvent e)
		{
			if (DateDblClick != null) {
				DateDblClick(sender, e);
			}
		}

		private void HandleGetDayBold(System.Object sender, AxMSComCtl2.DMonthViewEvents_GetDayBoldEvent e)
		{
			if (GetDayBold != null) {
				GetDayBold(sender, e);
			}
		}

		private void HandleSelChange(System.Object sender, AxMSComCtl2.DMonthViewEvents_SelChangeEvent e)
		{
			if (SelChange != null) {
				SelChange(sender, e);
			}
		}

		private void HandleClickEvent(System.Object sender, System.EventArgs e)
		{
			if (ClickEvent != null) {
				ClickEvent(sender, e);
			}
		}

		private void HandleDblClick(System.Object sender, System.EventArgs e)
		{
			if (DblClick != null) {
				DblClick(sender, e);
			}
		}

		private void HandleKeyDown(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyDown e)
		{
			if (KeyDown != null) {
				KeyDown(sender, e);
			}
		}

		private void HandleKeyUpEvent(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyUpEvent e)
		{
			if (KeyUpEvent != null) {
				KeyUpEvent(sender, e);
			}
		}

		private void HandleKeyPress(System.Object sender, AxMSComCtl2.DMonthViewEvents_KeyPress e)
		{
			if (KeyPress != null) {
				KeyPress(sender, e);
			}
		}

		private void HandleMouseDown(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseDown e)
		{
			if (MouseDown != null) {
				MouseDown(sender, e);
			}
		}

		private void HandleMouseMoveEvent(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseMoveEvent e)
		{
			if (MouseMoveEvent != null) {
				MouseMoveEvent(sender, e);
			}
		}

		private void HandleMouseUpEvent(System.Object sender, AxMSComCtl2.DMonthViewEvents_MouseUpEvent e)
		{
			if (MouseUpEvent != null) {
				MouseUpEvent(sender, e);
			}
		}

		private void HandleOLEStartDrag(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEStartDragEvent e)
		{
			if (OLEStartDrag != null) {
				OLEStartDrag(sender, e);
			}
		}

		private void HandleOLEGiveFeedback(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEGiveFeedbackEvent e)
		{
			if (OLEGiveFeedback != null) {
				OLEGiveFeedback(sender, e);
			}
		}

		private void HandleOLESetData(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLESetDataEvent e)
		{
			if (OLESetData != null) {
				OLESetData(sender, e);
			}
		}

		private void HandleOLECompleteDrag(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLECompleteDragEvent e)
		{
			if (OLECompleteDrag != null) {
				OLECompleteDrag(sender, e);
			}
		}

		private void HandleOLEDragOver(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEDragOverEvent e)
		{
			if (OLEDragOver != null) {
				OLEDragOver(sender, e);
			}
		}

		private void HandleOLEDragDrop(System.Object sender, AxMSComCtl2.DMonthViewEvents_OLEDragDropEvent e)
		{
			if (OLEDragDrop != null) {
				OLEDragDrop(sender, e);
			}
		}

	}
}

